### PR TITLE
Add two new OperationKinds for increment and decrement operators

### PR DESCRIPTION
**Pull request by mavasani**
Fixes https://github.com/dotnet/roslyn/issues/21748

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@mavasani It looks like there are legitimate test failures.

**Comment by mavasani** [comment]
@AlekseyTs Thanks - the test was not marked with Operation CompilerTrait attribute, so missed my local testing. Fixing and updating the PR now..

File: src/Compilers/Core/Portable/Operations/OperationKind.cs [hunk 1]
```diff
@@ -170,10 +170,12 @@ public enum OperationKind
         //PointerIndirectionReferenceExpression = 0x204,
         /// <summary>Indicates an <see cref="IIsPatternExpression"/>.</summary>
         IsPatternExpression = 0x205,
-        /// <summary>Indicates an <see cref="IIncrementExpression"/>.</summary>
+        /// <summary>Indicates an <see cref="IIncrementExpression"/> for increment operator.</summary>
         IncrementExpression = 0x206,
         /// <summary>Indicates an <see cref="IThrowExpression"/>.</summary>
         ThrowExpression = 0x207,
+        /// <summary>Indicates an <see cref="IIncrementExpression"/> for decrement operator.</summary>
+        DecrementExpression = 0x208,
```
Comment (cston) [inline src/Compilers/Core/Portable/Operations/OperationKind.cs]: Why is decrement an `IIncrementExpression`? Consider renaming `IIncrementExpression`, perhaps to `IIncrementOrDecrementExpression` if there is no better name.

Comment (mavasani) [reply src/Compilers/Core/Portable/Operations/OperationKind.cs]: @dotnet/analyzer-ioperation - I like Chuck's suggestion. Any objections?

**Review by AlekseyTs — APPROVED** [review]
LGTM (iteration 3)

**Review by cston — APPROVED** [review]
LGTM. Renaming `IIncrementExpression`, if needed, can be handled later.

**Review by jinujoseph — APPROVED** [review]

**Comment by mavasani** [comment]
@dotnet-bot retest windows_debug_unit64_prtest please

**Comment by mavasani** [comment]
@dotnet-bot retest windows_release_unit64_prtest please

**Pull request merged**